### PR TITLE
[3/n][pipeline-gen] Add TestStep and BuildkiteStep along with validators

### PR DESCRIPTION
**Pull request by khluu**
- Add these 2 step classes to be used in future PRs
- Add validators for `TestStep`

### CODE REVIEW

File: scripts/pipeline_generator/step.py [hunk 1]
```diff
@@ -1,9 +1,72 @@
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, Field, root_validator, model_validator
 from typing import List, Dict, Any, Optional
-
-from .utils import AgentQueue
+from typing_extensions import Self
+from .utils import AgentQueue, GPUType
```
Comment (aslonnie) [inline scripts/pipeline_generator/step.py:4]: keep the separate line between local and third-party imports?

File: scripts/pipeline_generator/step.py [hunk 2]
```diff
@@ -1,9 +1,72 @@
...
+            raise ValueError("'num_gpus' must be defined if 'num_nodes' is defined.")
+        if self.num_nodes and len(self.commands) != self.num_nodes:
+            raise ValueError("Number of commands must match the number of nodes.")
+        return self
+
+
+class BuildkiteStep(BaseModel):
+    """This class represents a step in Buildkite format."""
+    label: str
+    key: str
```
Comment (aslonnie) [inline scripts/pipeline_generator/step.py:60]: `key` is always required?

**Review by aslonnie — APPROVED** [review]

File: scripts/pipeline_generator/step.py (hunk 1 above)
Comment (khluu) [reply scripts/pipeline_generator/step.py:5]: done

File: scripts/pipeline_generator/step.py (hunk 2 above)
Comment (khluu) [reply scripts/pipeline_generator/step.py]: oh `key` is not required

Comment (khluu) [reply scripts/pipeline_generator/step.py]: I made it optional now

**Pull request merged**